File: transport_service_commands/src/main/java/pl/edu/pg/transport/entity/Flight.java
package pl.edu.pg.transport.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "flights")
@NoArgsConstructor
@Getter
public class Flight {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;

    @Column(name = "departure_airport")
    private String departureAirport;

    @Column(name = "arrival_airport")
    private String arrivalAirport;

    @Column(name = "departure_date")
    private LocalDateTime departureDate;

    @Column(name = "arrival_date")
    private LocalDateTime arrivalDate;

    @Column(name = "travel_time")
    private int travelTime;

    @Column(name = "places_count")
    private int placesCount;

    @Column(name = "places_occupied")
    private int placesOccupied;

    public Flight(String departureAirport, String arrivalAirport,
                  LocalDateTime departureDate, LocalDateTime arrivalDate,
                  int travelTime, int placesCount) {
        this.departureAirport = departureAirport;
        this.arrivalAirport = arrivalAirport;
        this.departureDate = departureDate;
        this.arrivalDate = arrivalDate;
        this.travelTime = travelTime;
        this.placesCount = placesCount;
        this.placesOccupied = 0;
    }
}

File: tour_operator/flights/Flight.py
import json


class Flight:
    def __init__(self, flight):
        self.id = None
        if "id" in flight:
            self.id = flight["id"]
        self.departureAirport = flight["departureAirport"]
        self.arrivalAirport = flight["arrivalAirport"]
        self.departureDate = flight["departureDate"]
        self.arrivalDate = flight["arrivalDate"]
        self.travelTime = flight["travelTime"]
        self.placesCount = flight["placesCount"]
        self.price = flight["price"]

    def compare(self, flight):
        if not isinstance(flight, str):
            return False

        flight_dict = json.loads(flight)
        if self.departureAirport != flight_dict["departureAirport"]:
            return False
        if self.arrivalAirport != flight_dict["arrivalAirport"]:
            return False
        if self.departureDate != flight_dict["departureDate"]:
            return False
        if self.arrivalDate != flight_dict["arrivalDate"]:
            return False
        if self.travelTime != flight_dict["travelTime"]:
            return False

        return True
